### PR TITLE
CORE-13740: Add serial and allow conflicts in suspension and activation

### PR DESCRIPTION
**Pull request by yift-r3**
See the e2e pull request in https://github.com/corda/corda-e2e-tests/pull/109

### CODE REVIEW

File: testing/e2e-test-utilities/src/main/kotlin/net/corda/e2etest/utilities/MGMUtils.kt [hunk 1]
```diff
@@ -309,18 +309,19 @@ private fun ClusterInfo.createMgmRegistrationContext(
...
+    serialNumber: Int? = null,
 ) = cluster {
     assertWithRetry {
         timeout(15.seconds)
         interval(1.seconds)
         command {
             post(
                 "/api/v1/mgm/$mgmHoldingId/suspend",
-                "{ \"x500Name\": \"$x500Name\" }"
+                "{ \"x500Name\": \"$x500Name\", \"serialNumber\": $serialNumber }"
```
Comment (dimosr) [inline testing/e2e-test-utilities/src/main/kotlin/net/corda/e2etest/utilities/MGMUtils.kt:321]: Does the REST endpoint work fine with if `serialNumber` is null? If so, should we omit specifying it at all in cases it's null?

 I remember there was an issue with blank strings.

**Review by dimosr — COMMENTED** [review]
LGTM - just a minor comment

File: testing/e2e-test-utilities/src/main/kotlin/net/corda/e2etest/utilities/MGMUtils.kt (hunk 1 above)
Comment (williamvigorr3) [reply testing/e2e-test-utilities/src/main/kotlin/net/corda/e2etest/utilities/MGMUtils.kt:321]: Hmm I don't think `"{ \"x500Name\": \"$x500Name\", \"serialNumber\": null }"` is valid json, so we should probably fix this.

**Review by williamvigorr3 — COMMENTED** [review]
Everything apart from the open comment looks fine.

File: testing/e2e-test-utilities/src/main/kotlin/net/corda/e2etest/utilities/MGMUtils.kt (hunk 1 above)
Comment (yift-r3) [reply testing/e2e-test-utilities/src/main/kotlin/net/corda/e2etest/utilities/MGMUtils.kt:321]: The rest endpoint work with null as if the field is not there.

(The issue with blank string is that it behaves as null)

Comment (yift-r3) [reply testing/e2e-test-utilities/src/main/kotlin/net/corda/e2etest/utilities/MGMUtils.kt:321]: (Am I missing something? It looks like perfectly valid JSON to me)

Comment (williamvigorr3) [reply testing/e2e-test-utilities/src/main/kotlin/net/corda/e2etest/utilities/MGMUtils.kt:321]: Yeah I think you are correct (having read the json spec).

**Review by dimosr — APPROVED** [review]

**Pull request merged**